configs/components/runtime-pe-bolt-server.rb

```diff
@@ -19,4 +22,22 @@
   end
 
   pkg.build_requires('puppet-agent')
-end
\ No newline at end of file
+  if builtin_ruby
+    pkg.build_requires "libffi"
+    pkg.build_requires "libyaml"
+
+    # PROJECT_SHORTNAME is used in the runtime script to determine the
+    # libdir where we put all the linked lib files. For bolt-server there's
+    # a whole path it needs to follow instead of just a name like 'installer'
+    # or 'bolt'.
+    pkg.environment "PROJECT_SHORTNAME", "server/apps/bolt-server"
+    pkg.add_source "file://resources/files/runtime/runtime.sh"
+    if platform.name =~ /el-[567]|redhatfips-7|sles-(:?11|12)|ubuntu-18.04-amd64/
```

Comment:
what is the `:?` in the sles portion of the regex doing?

Reply:
I.... don't know....

This was copied from the agent's runtime definition, and looking back at the blame it was a change from Ciprian and there's no explanation as to why this includes `:?`

Maybe it was supposed to be a non-capturing group? Which still doesn't make sense but at least `?:` is an actual thing in regex language. Anyway I guess it _technically_ captures `sles-:11` and `sles-11` but I have nfc why that would be important.

Reply:
imo, we should remove it